doc/ag.txt

```diff
@@ -138,16 +138,36 @@ window is opened, or what size it is. Example: >
   let g:ag_qhandler="copen 20"
 <
 
+                                                           *g:ag_lmapping_func*
+A custom command used to set key mappings for the location list. Default:
+"call ag#AgSetDefaultMappings('l')". Example: >
+  let g:ag_qhandler="call AgSetMyCustomMappings()"
+<
+
+
+                                                           *g:ag_qmapping_func*
+A custom command used to set key mappings for the quickfix window. Default:
```

Comment:
You could have three optional sections. Global mappings, location list
mappings, and quickfix mappings. The most specific mappings would take
precedence, falling back to default values if we didn't find a specific
mapping or group of mappings.
On Oct 14, 2015 19:27, "Sai Kalidindi" <notifications@github.com> wrote:

> In doc/ag.txt
> <https://github.com/rking/ag.vim/pull/113#discussion_r42071618>:
>
> > @@ -138,16 +138,36 @@ window is opened, or what size it is. Example: >
> >    let g:ag_qhandler="copen 20"
> >  <
> >
> > +                                                           *g:ag_lmapping_func*
> > +A custom command used to set key mappings for the location list. Default:
> > +"call ag#AgSetDefaultMappings('l')". Example: >
> > +  let g:ag_qhandler="call AgSetMyCustomMappings()"
> > +<
> > +
> > +
> > +                                                           *g:ag_qmapping_func*
> > +A custom command used to set key mappings for the quickfix window. Default:
>
> I'm not against doing that, but wouldn't that require two keymaps? One for
> quickfix window mode and one for location list mode? The current
> implementation allows for a function that can take arguments and avoid
> duplicate code.
>
> Again, I'm not against switching over to something like that, since
> ack.vim seems to do a similar thing. I'm actually interested in the
> logistics.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rking/ag.vim/pull/113/files#r42071618>.
>